src/mscorlib/src/System/String.cs

```diff
@@ -573,10 +572,13 @@ public override bool Equals(Object obj)
         // Determines whether two strings match.
         [Pure]
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I am not sure whether it is a good trade-off to mark this with AggressiveInlining. String.Equals is called in a lot of places. Bloating the code for all of them can easily out-weight the gains from the inlining. The empirical rule is that optimizing for smaller code by default gives better overall performance.